lib/rubocop/cop/rspec/file_path.rb

```diff
@@ -131,11 +140,9 @@ def ignore_methods?
           cop_config['IgnoreMethods']
         end
 
-        def filename_ends_with?(glob)
-          filename =
-            RuboCop::PathUtil.relative_path(processed_source.buffer.name)
-              .gsub('../', '')
-          File.fnmatch?("*#{glob}", filename)
+        def filename_ends_with?(pattern)
+          filename = File.expand_path(processed_source.buffer.name)
+          filename.match?(".*#{pattern}$")
```

Comment:
`.*` is unnecessary at the beginning (or end) of a regex.

Reply:
Could we move the `$` (which should probably be `\z`?) into the pattern itself? Then this line would become simply `filename.match?(pattern)`…

Reply:
- You are right about unnecessary `.*` in the beginning. It is removed.
- We cannot move the `$` or `\z` to the `pattern_for` method because it would then appear in the offense messages unless it is also substituted. I think that it makes sense to keep the ending part of the pattern in the filename_**ends_with?** method.
- Regarding `$` or `\z` I guess either one is fine. Is there any case when a single filename can be multiline?